src/hw_isolation_record/manager.cpp

```diff
@@ -553,9 +564,17 @@ void Manager::restore()
     // by BMC and Hostboot
     openpower_guard::GuardRecords records = openpower_guard::getAll(true);
 
-    std::for_each(records.begin(), records.end(), [this](const auto& record) {
+    auto validRecord = [this](const auto& record) {
+        return this->isValidRecord(record.recordId);
```

Comment:
you could have added the implementation of isValidRecord in the lambda itself but ok your choice

Reply:
Yeah, I thought about it but, I would like to leave it as the utils function so that we can use future if needed outside of the lambda too (just for checking the given record id).